src/initilization-context.ts

```diff
@@ -30,9 +30,11 @@ export class InitializationContext {
 	wait(task: Promise<any>) {
 		this.tasks.add(task);
 		task.then(() => {
-			this.tasks.delete(task);
 			// process the queue asynchronously to allow additional tasks to be queued as a result of this one
```

Comment:
As you can see in this comment, the original intention here was to ensure any tasks that depended on this task would be able to run _before_ we considered this task to be truly complete.

The specific action that raised the issue was [here](https://github.com/cognitoforms/model.js/blob/ef98f9ff3bcb83f249cce13b2e5a3cbb2fc56e42/src/entity.ts#L220).

The `then()` callback which deletes `task` from the task queue was guaranteed to run before any callbacks from the caller's frame, introducing the possibility that the "waiting queue" could be processed prematurely.